### DISCUSSION TITLE
Why the menu closing animation has unhandled promise reject?

### DISCUSSION THREAD

**Discussion by VietNguyenQuoc** [material-components/material-web#5638] [Q&A] [answered]
I don't know if it's a bug or some intended implementation.
I'm wrapping the menu with React and sometimes it crashes when closing the menu with animation (`quick=true` works)

Diving into the material code, I see that it will reject a promise in several scenarios:
- no anchor or surface elements
- there's another closing animation fired after the current one.

Can someone explain why the menu closing animation has to reject the promise [in here](https://github.com/material-components/material-web/blob/main/menu/internal/menu.ts#L873)  and why the promise rejection is not handled [in here?](https://github.com/material-components/material-web/blob/main/menu/internal/controllers/surfacePositionController.ts#L613-L616)

**Comment by VietNguyenQuoc** [comment]
@asyncLiz @e111077 can you help with this?

**Reply by asyncliz** [reply, marked as answer]
At a glance I don't think I see a reason for `animateClose()` on the menu to reject the promise. I think it can just resolve early instead of rejecting when there aren't elements or when animations are canceled.

@e111077 do you agree? If so that should be a simple PR we can accept.

**Reply by e111077** [reply]
probably should resolve early. Wanted to make sure there was some way to determine the difference between whether it succeeded or failed (can probably just do `resolve(false)`), but likely YAGNI.

**Reply by asyncliz** [reply]
Thanks Elliott! @VietNguyenQuoc could you send a PR that changes the rejections to resolve instead?

**Reply by VietNguyenQuoc** [reply]
thanks @asyncLiz @e111077, I'll send the PR.

**Reply by VietNguyenQuoc** [reply]
Oh I see @e111077 already done that.
